index.d.ts

```diff
@@ -274,6 +274,14 @@ declare module '@primer/components' {
   export const UnderlineNav: React.FunctionComponent<UnderlineNavProps>
 
   export const theme: Object
+
+  export interface DialogProps extends CommonProps {
+    title: string
```

Comment:
`title` [can also be a React Node](https://github.com/primer/components/blob/master/src/Dialog.js#L106), but making it `string | React.ReactNode` causes an error since that's not compatible with the `CommonProps` `title` attribute.

To solve this I would need to change quite a bit the type definitions of `CommonProps`, `BaseProps` and probably the props of most of the components, let me know if it's worth it.

Reply:
@mxstbr do you know why `title` needs to be in `CommonProps`? https://github.com/primer/components/blob/master/index.d.ts#L9

Reply:
:wave: Back from vacation now—probably because `a` tags can have a string `title` I would assume? That is copied from the rebass types if I remember correctly.

Reply:
@emplums I guess then that this PR should be fine to merge, and as a follow-up we can do the refactor of `Props` to support the `title` of a `Dialog` to be a React Node.